client/src/ui/ToasterWebview.ts

```diff
@@ -0,0 +1,117 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) 2023 Savoir-faire Linux. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as vscode from 'vscode'
+
+const DEFAULT_TOASTER_PORT = 8000
+export class ToasterPanel {
+  /**
+  * Track the currently panel. Only allow a single panel to exist at a time.
+  */
+  public static currentPanel: ToasterPanel | undefined
+
+  public static readonly viewType = 'yoctoToaster'
+
+  private readonly _panel: vscode.WebviewPanel
+  private readonly _disposables: vscode.Disposable[] = []
+
+  public static createOrShow (): void {
+    const column = vscode.window.activeTextEditor !== undefined
+      ? vscode.window.activeTextEditor.viewColumn
+      : undefined
+
+    // If we already have a panel, show it.
+    if (ToasterPanel.currentPanel !== undefined) {
+      ToasterPanel.currentPanel._panel.reveal(column)
+      return
+    }
+
+    // Otherwise, create a new panel.
+    const panel = vscode.window.createWebviewPanel(
+      ToasterPanel.viewType,
+      'Yocto Toaster',
+      column ?? vscode.ViewColumn.One,
+      getWebviewOptions()
+    )
+
+    ToasterPanel.currentPanel = new ToasterPanel(panel)
+  }
+
+  private constructor (panel: vscode.WebviewPanel) {
+    this._panel = panel
+
+    // Set the webview's initial html content
+    this._setInitialView()
+
+    // Listen for when the panel is disposed
+    // This happens when the user closes the panel or when the panel is closed programmatically
+    this._panel.onDidDispose(() => { this.dispose() }, null, this._disposables)
+
+    // Update the content based on view changes
+    this._panel.onDidChangeViewState(
```

Comment:
Is it mandatory to implement it? Since we don't do anything in it.

Reply:
Sure, we could discard it if there is nothing to put there.